llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -58,6 +56,26 @@ void llvm::getSYCLDeviceRequirements(
         std::vector<uint32_t>(Values.begin(), Values.end());
   }
 
+  std::optional<llvm::SmallVector<int64_t, 3>> ReqdWorkGroupSize;
+  for (const Function &F : MD.getModule()) {
+    if (const MDNode *MDN = F.getMetadata("reqd_work_group_size")) {
+      llvm::SmallVector<int64_t, 3> NewReqdWorkGroupSize;
+      for (const auto &operand : MDN->operands())
+        NewReqdWorkGroupSize.push_back(
+            ExtractIntegerFromMDNodeOperand(operand));
+      if (!ReqdWorkGroupSize)
+        ReqdWorkGroupSize = NewReqdWorkGroupSize;
+      else if (!std::equal(ReqdWorkGroupSize->begin(), ReqdWorkGroupSize->end(),
+                           NewReqdWorkGroupSize.begin())) {
+        // two functions in the module have different required work group sizes
+      }
```

Comment:
If the branch is empty, then we don't need it at all. We could add an assert there, but if we do so, I would also suggest to put the whole branch under something like `#ifndef NDEBUG`